File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationCenterInParent.java
package com.github.florent37.expectanim.core.position;

import android.view.View;
import android.view.ViewParent;

/**
 * Created by florentchampigny on 17/02/2017.
 */

public class PositionAnimExpectationCenterInParent extends PositionAnimExpectation {

    public boolean horizontal;
    public boolean vertical;

    public PositionAnimExpectationCenterInParent(boolean horizontal, boolean vertical) {
        this.horizontal = horizontal;
        this.vertical = vertical;

        setForPositionX(true);
        setForPositionY(true);
    }

    @Override
    public Float getCalculatedValueX(View viewToMove) {
        final ViewParent viewParent = viewToMove.getParent();
        if((viewParent instanceof View) && horizontal){
            final View parentView = (View)viewParent;
            return parentView.getWidth() / 2f - viewToMove.getWidth() / 2f;
        }
        return null;
    }

    @Override
    public Float getCalculatedValueY(View viewToMove) {
        final ViewParent viewParent = viewToMove.getParent();
        if((viewParent instanceof View) && vertical){
            final View parentView = (View)viewParent;
            return parentView.getHeight() / 2f - viewToMove.getHeight() / 2f;
        }
        return null;
    }
}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/position/PositionAnimExpectationCenterInParent.kt
package com.github.florent37.kotlin.pleaseanimate.core.position

import android.view.View

class PositionAnimExpectationCenterInParent(var horizontal: Boolean, var vertical: Boolean) : PositionAnimExpectation() {

    init {
        isForPositionX = true
        isForPositionY = true
    }

    override fun getCalculatedValueX(viewToMove: View): Float? {
        val viewParent = viewToMove.parent
        if (viewParent is View && horizontal) {
            val parentView = viewParent as View
            return parentView.width / 2f - viewToMove.width / 2f
        }
        return null
    }

    override fun getCalculatedValueY(viewToMove: View): Float? {
        val viewParent = viewToMove.parent
        if (viewParent is View && vertical) {
            val parentView = viewParent as View
            return parentView.height / 2f - viewToMove.height / 2f
        }
        return null
    }
}
